src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,99 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPASITY = 16;
+    private static final float DEFAULT_LOAD_FACTORY = 0.75f;
+    private int sizeCounter;
+    private int capacity;
+    private Node<K, V>[] elementsTable;
+
+    public MyHashMap() {
+        sizeCounter = 0;
+        capacity = DEFAULT_CAPASITY;
+        elementsTable = new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
+        if (sizeCounter >= capacity * DEFAULT_LOAD_FACTORY) {
+            resize();
+        }
 
+        int index = indexByKeyHash(key);
+        if (elementsTable[index] == null) {
+            elementsTable[index] = new Node<>(key, value, null);
+            sizeCounter++;
+            return;
+        }
```

Comment:
do we need this chacking?
let's remove it and move logic from method `addToNode` here